docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,325 @@
+```
+
+### oras push
+
+Push two files to a repository and show the descriptor of the image manifest in pretty JSON format. 
+
+```bash
+oras push $REGISTRY/$REPO:$TAG sbom.spdx vul-scan.json --format json 
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "MediaType": "application/vnd.oci.image.manifest.v1+json",
+  "Digest": "sha256:4a5b8c83d153f52afdfcb422db56c2349aae3bd5ecf8338a58353b5eb6681c45",
+  "Size": 820,
+  "Annotations": {
+    "org.opencontainers.image.created": "2023-12-15T09:41:54Z"
+  }
+}
+```
+
+> [!NOTE]
+> When pushing a folder to filesystem, the output fields are the same as the example of pushing files as above.
+
+Push a folder to a repository and filter out the value of `reference` and `artifactType` of the pushed artifact in the standard output.
+
+```bash
+oras push $REGISTRY/$REPO:$TAG sample-folder --format '{{.Ref}}, {{.MediaType}}'
+```
+
+```console
+localhost:5000@sha256:85438e6598bf35057962fff34399a362d469ca30a317939427fca6b7a289e70d, application/vnd.oci.image.manifest.v1+json
+```
+
+### oras attach
+
+Attach two files to an image and show the descriptor metadata of the referrer in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/report-and-sbom vul-report.json:example/vul-scan sbom.spdx:example/sbom --format json
+```
+
+```json
+{
+  "Ref": "localhost:5000/oras@sha256:0afd0f0c35f98dcb607de0051be7ebefd942eef1e3a6d26eefd1b2d80f2affbe",
+  "MediaType": "application/vnd.oci.image.manifest.v1+json",
+  "Digest": "sha256:0afd0f0c35f98dcb607de0051be7ebefd942eef1e3a6d26eefd1b2d80f2affbe",
+  "Size": 923,
+  "Annotations": {
+    "org.opencontainers.image.created": "2023-12-15T08:59:21Z"
+  }
+}
+```
+
+### oras discover
+
+View an artifact's referrers. The default output should be listed in a tree view.
+
+```bash
+oras discover $REGISTRY/$REPO:$TAG --format tree
+```
+
+```console
+localhost:localhost:5000/hello@sha256:5cb894d0c94c56894e160ad2eeb19a123b4d2155374e7709f43f8c0c2f249fe2
+├── application/vnd.oci.empty.v1+json
+│   └── sha256:0db683b656132cede5360f42bc52541f3386b30ce685e6e63ff93ced54423fb8
+└── application/vnd.cncf.notary.signature
+    └── sha256:476d43120d3799fa76d7706f741beca73f5ff4149c8b6db3bd516a73d4a82fc1
+```
+
+View an artifact's referrers manifest in pretty JSON output. The following fields should be outputted:
+
+- `Manifests`: the list of referrers
+  - `Ref`: full reference by digest of the referrer
+  - `MediaType`: media type of the referrer
+  - `Size`: referrer file size in bytes
+  - `Digest`: digest of the referrer
+  - `ArtifactType`: artifact type of a referrer
+  - `Annotations`: contains arbitrary metadata in a referrer
+
+See an example:
+
+```bash
+oras discover localhost:5000/hello:v1 --format json
+```
+
```

Comment:
same feedback as before. I would recommend moving all these to lower case properties.

Reply:
Replied above